scripts/gceworker.sh

```diff
@@ -168,6 +168,17 @@ case "${cmd}" in
     shift
     gcloud compute scp --recurse "${from}" "$@"
     ;;
+    put)
+    lpath="${1}"
+    rpath="${2}"
+    # Check whether we have an absolute path like /foo, ~foo, or ~/foo.
+    # If not, base the path relative to the CRDB repo.
+    if [[ "${rpath:0:1}" != / && "${rpath:0:2}" != ~[/a-z] ]]; then
+        rpath="go/src/github.com/cockroachdb/cockroach/${rpath}"
```

Comment:
This is fairly surprising behavior IMO. I'm wondering what problem this logic is supposed to solve and why we seem to go out of our way to do something surprising.

Personally, as a person using a script, I expect that a non-absolute path is a relative path, and is relative to my current working directory. I would absolutely 100% of the time NEVER expect the path to be relative to some random directory on my machine. For me I think this will result in the script appearing to not find a file that I know exists, or, even worse, choosing the wrong file than the one I meant.

Also, you can have multiple `cockroach` checkouts, and it doesn't necessarily have to be at `/go/src/github.com/cockroachdb/cockroach`.

Reply:
`~` is also expanded by the shell. You shouldn't have to check for it manually.

Reply:
Great points. Looks like this logic was added to `get` in https://github.com/cockroachdb/cockroach/pull/80723 and https://github.com/cockroachdb/cockroach/pull/97177. It seems strange to me, too, so I'll remove it from `put`.